server/controllers/profile.js

```diff
@@ -1,97 +1,117 @@
+const mongoose = require("mongoose");
 
-const Profile = require('../models/Profile');
+const Profile = require("../models/Profile");
 
 const asyncHandler = require("express-async-handler");
 
-// @route POST 
+// @route POST /profile/new
 // Create New User Profile
 exports.newUser = asyncHandler(async (req, res) => {
-    const { userId, firstName, lastName, description, location, isDogOwner, isDogSitter, rating, hourlyRate, tagLine } = req.body;
-    
-    const profile = await Profile.create({
-        userId,
-        firstName,
-        lastName,
-        description,
-        location,
-        isDogOwner,
-        isDogSitter,
-        rating,
-        hourlyRate,
-        tagLine
-    });
+  const {
+    userId,
+    firstName,
+    lastName,
+    description,
+    location,
+    images,
+    isDogSitter,
+    availabilityWeek,
+    availabilityDays,
 
-    if (profile) {
-         res.status(201).json({
-            userId: profile.userId,
-            firstName: profile.FirstName,
-            lastName: profile.lastName,
-            description: profile.description,
-            location: profile.location,
-            isDogSitter: profile.isDogSitter,
-            rating: profile.rating,
-            hourlyRate: profile.hourlyRate,
-            tagLine: profile.tagLine,
-        })
-    } else {
-       res.status(400).json({ message: "Invalid User Data"})
-    }
+    rating,
+    hourlyRate,
+    tagLine,
+  } = req.body;
+
+  const profile = await Profile.create({
+    userId,
+    firstName,
+    lastName,
+    description,
+    location,
+    images,
+    isDogSitter,
+    availabilityWeek,
+    availabilityDays,
+    rating,
+    hourlyRate,
+    tagLine,
+  });
+
+  if (profile) {
+    res.status(201).json({
+      userId: profile.userId,
+      firstName: profile.FirstName,
+      lastName: profile.lastName,
+      description: profile.description,
+      location: profile.location,
+      images: profile.images,
+      isDogSitter: profile.isDogSitter,
+      availabilityWeek: profile.availabilityWeek,
+      availabilityDays: profile.availabilityDays,
+      rating: profile.rating,
+      hourlyRate: profile.hourlyRate,
+      tagLine: profile.tagLine,
+    });
+  } else {
+    res.status(400).json({ message: "Invalid User Data" });
+  }
 });
 
-// //@route UPDATE
-// //update profiles 
+// //@route Patch /profile/editprofile/:id
+// //update profiles
 exports.editProfile = asyncHandler(async (req, res) => {
-    const userId = req.params.id
-    let profile;
-    try {
-        const updateProfile = await Profile.findOneAndUpdate({userId: userId});
-            if(updateProfile) { 
-                profile.save(res.status(200).json({profile: updateProfile})
-            )} else {
-                res.status(404).json({ message: "Profile not found"})
-            }
-        } catch (error) {
-            return res.status(500).json({error: "Could not update profile"})
-        };
+  const userId = req.params.id;

```

Comment:
for this controller, will we ever need `/editProfile` for other users? meaning will other users be able to edit other user's profiles?

from my assumptions, likely not. in which we should not open this endpoint to the public. We'll remove `/:id` from the controller and access the logged in user ID through cookies instead :)